### ISSUE TITLE
PiecewiseChebyshev does not work as expected when breakpoints = None

### ISSUE THREAD

**Issue by EgrettaThula** [Qiskit/qiskit-terra#6198]
### Information

- **Qiskit Terra version**: Master
- **Python version**: 3.7.5
- **Operating system**: Windows 10

### What is the current behavior?
When using PiecewiseChebyshev  with breakpoints equal None, the resulting circuit contains one gate only.

### Steps to reproduce the problem
```python
f_x, degree, breakpoints, num_state_qubits = lambda x: np.sqrt(x), 2, None, 2
pw_approximation = PiecewiseChebyshev(f_x, degree, breakpoints, num_state_qubits)
print(pw_approximation.breakpoints)
print(pw_approximation.polynomials)
```
Result:
```
[0, 4]
[[3.141592653589793]]
```
### What is the expected behavior?
Result should be:
```
[0, 4]
[[0.6901841202733885, 1.321746423545076, -0.12631246065433754]]
```

### Suggested solutions
I think ```self.breakpoints``` should be used instead of ```self._breakpoints``` in ```polynomials``` property

**Comment by Cryoris** [comment]
@anedumla could you have a look? 🙂

**Comment by Cryoris** [comment]
I'll close this as this should've been fixed by #6210.
